packages/boxel-ui/addon/src/components/resizable-panel-group/index.gts

```diff
@@ -595,65 +525,41 @@ export default class ResizablePanelGroup extends Component<Signature> {
         (panelRatio) => panelRatio / totalNewPanelRatio,
       );
 
-      panelContexts.forEach((panelContext, index) => {
+      panels.forEach((panel, index) => {
         let panelRatio = newPanelRatios[index];
         if (!panelRatio) {
           console.warn('Expected panelRatio to be defined');
           return;
         }
         let proportionalSize = panelRatio * remainingContainerSize;
         let actualSize = Math.round(proportionalSize);
-        panelLengths[panelContext.id] = actualSize;
+
+        panelLengths[index] = actualSize;
+        panelToNewLength.set(panel, actualSize);
       });
     };
-    calculateLengthsOfPanelWithoutMinLength();
-
-    for (let index = 0; index <= this.listPanelContext.size; index++) {
-      let panelContext = this.listPanelContext.get(index);
-      if (panelContext) {
-        this.listPanelContext.set(index, {
-          ...panelContext,
-          lengthPx: panelLengths[index] || 0,
-        });
-      }
-    }
+    calculateLengthsOfPanelsWithoutMinLength();
```

Comment:
I recall this is the main calculation code that consolidates length and ratio and the panels. Its relatively complicated.  And I know thereis some issues with this code bcos ratios are normalised as in between of adding and removing new panels. This will essentially, lead to ratios being morphed. For example, when going from 3 panels to 2 panels by removing panels and adding

It goes like
- 3 panels 0.2,0.4,0.4 
- 2 panels 0.33, 0..66 (unregistered a panel)
- 1 panel 1 (unregistered a panel) 
- 2 panel 0.769, 0.23 (registered a panel) 

You can see that the ratios morphed maybe not for the better 

In the spirit to be aware of this, can we separate this into a block of code where ratio calculation exist. Then we can maybe add test there in the future

Reply:
yes, I see what you’re saying. I’ll include something like this in the followup that addresses the ratios. Does this mean that maybe the old ratios before hiding should be tracked and restored when unhidden? Seems like an explosion of complexity is possible 😳

Reply:
It sounds like an explosion of complexity if its all inside the component. Tho, im also skeptical that you can prevent the ratios from being morphed if you commit to the principle of recalculate them every time you register or unregister a new panel. 

we probably have to decide on a heuristic for this to be stable.  But, I think its just hard to think about at its current state. For hiding and persistence, I didn't think you need to change the ratios. I thought you'd just store some new state in the local storage?

Reply:
I think this code should belong together in some sort of abstraction. wdyt

- `calculateLengthsOfPanelsWithMinLength` , `calculateLengthsOfPanelsWithoutMinLength` uses the intermediate ratios from registering to update the lengths onContainerResize
- `calculateRatios` updates the intermediate ratios upon register, unregistering, moving the handles
 
tho im ok also not including these changes in this PR